File: SpotifyBrowser.Backend/SpotifyBrowser.ReadStack.Api.Host/MusicModels/Track.cs
using System.Collections.Generic;

namespace SpotifyBrowser.ReadStack.Api.Host.MusicModels
{
    public class Track
    {
        public Album Album { get; set; }
        public List<Artist> Artists { get; set; }
        public List<string> AvailableMarkets { get; set; }
        public int DiscNumber { get; set; }
        public int DurationMs { get; set; }
        public string Href { get; set; }
        public string Id { get; set; }
        public string Name { get; set; }
        public string PreviewUrl { get; set; }
        public int TrackNumber { get; set; }
        public string Type { get; set; }
        public string Uri { get; set; }
    }
}

File: spotify-browser-frontend/src/app/core/music/model/track.ts
import { Artist } from './artist';
import { Album } from './album';

export interface Track {
    album: Album;
    artists: Artist[];
    availableMarkets: string[];
    discNumber: number;
    durationMs: number;
    href: string;
    id: string;
    name: string;
    previewUrl?: any;
    trackNumber: number;
    type: string;
    uri: string;
}
